pymobiledevice3/services/crash_reports.py

```diff
@@ -127,28 +128,36 @@ def watch(self, name: str = None, raw: bool = False) -> Generator[str, None, Non
                 else:
                     yield crash_report
 
-    def get_new_sysdiagnose(self, out: str, erase: bool = True) -> None:
+    def get_new_sysdiagnose(self, out: str, erase: bool = True, *, timeout: Optional[float] = None) -> None:
         """
         Monitor the creation of a newly created sysdiagnose archive and pull it
         :param out: filename
         :param erase: remove after pulling
+        :keyword timeout: Maximum time in seconds to wait for the completion of sysdiagnose archive
+            If None (default), waits indefinitely
         """
-        sysdiagnose_filename = self._get_new_sysdiagnose_filename()
+        if timeout is None:
+            timeout = float('inf')
+        timeout = time.time() + timeout
```

Comment:
this name is confusing since this is not a timeout interval

Reply:
created new variable `end_time`